lib/ast/expression2.ts

```diff
@@ -74,6 +88,7 @@ export abstract class Expression extends Node {
     abstract toLegacy(into_params ?: InputParam[], scope_params ?: string[]) : legacy.Stream|legacy.Table|legacy.Action;
     abstract clone() : Expression;
     abstract toSource() : TokenStream;
+    abstract equals(other : Expression) : boolean;
```

Comment:
This is quite useful, because it means we can clean up the equals implementation in DialogueStateHistoryItem